shakenfist/external_api/instance.py

```diff
@@ -243,8 +244,38 @@ def post(self, name=None, cpus=None, memory=None, network=None, disk=None,
             if 'memory' not in video:
                 return api_base.error(400, 'video specification requires "memory"')
 
+        # Verify tags
+        if tags:
+            if not isinstance(tags, list):
+                return api_base.error(400, 'tags should be a list of strings')
+            for t in tags:
+                if not isinstance(t, str):
+                    return api_base.error(400, 'tag (%s) is not a string' % t)
+
+        # Verify affinity settings
+        if affinity:
+            if not isinstance(affinity, dict):
+                return api_base.error(400, 'affinity should be a dictionary')
+
+            for key_type, aff in affinity.items():
+                # Check the validity of each affinity setting
+                if key_type not in ('cpu', 'disk', 'instance'):
```

Comment:
Can I specify an affinity with an instance in a different namespace?

Reply:
Maybe [he says coyly]. This would be very useful but depends on whether all users are trusted. Maybe make that a installation configuration option.

This is probably a bigger picture decision, whether the whole installation should have this an installation option?

Reply:
We've talked before about a heirarchy of namespaces, and I think this falls into that future looking category to be honest. I'd have a namespace for DB servers for example, and then I could say "I don't like sharing with that namespace" somewhere else?

Reply:
That's a good point. If we implement the hierarchy of namespaces, we could specify which namespaces can leak information between each other.

So, we currently stop interaction between namespaces for other purposes. This PR should (for now) be the same. Concur?

Reply:
I agree for now. I see the leakage between namespaces being expressed by the heirarchy -- if you have a common owner other than system, you can share.